src/assignment/modals/AddBlock.tsx

```diff
@@ -0,0 +1,128 @@
+
+interface AddBlockProps extends Pick<ModalProps, 'isOpen' | 'onClose'> {
+	assignment: AssignmentFormState;
+	onConfirm?: (type: AddBlockType) => void;
+}
+
+const AddBlock: FunctionComponent<AddBlockProps> = ({ assignment, isOpen, onClose, onConfirm }) => {
+	const [t] = useTranslation();
+
+	const items: {
+		type: AddBlockType;
+		icon: IconNameSchema;
+		title: ReactNode;
+		description: ReactNode;
+		disabled?: boolean;
+	}[] = useMemo(
+		() => [
+			{
+				type: AssignmentBlockType.ITEM,
+				icon: EDIT_ASSIGNMENT_BLOCK_ICONS()[AssignmentBlockType.ITEM],
+				title: t('assignment/modals/add-block___kijken-luisteren-fragment'),
+				description: t(
+					'assignment/modals/add-block___voeg-een-fragment-uit-je-werkruimte-toe-om-te-laten-bekijken-of-beluisteren'
+				),
+			},
+			{
+				type: 'COLLECTIE',
+				icon: 'collection',
+				title: t('assignment/modals/add-block___kijken-luisteren-collectie'),
+				description: t(
+					'assignment/modals/add-block___start-je-opdracht-vanaf-een-bestaande-collectie-fragmenten-uit-je-werkruimte'
+				),
+			},
+			{
+				type: AssignmentBlockType.ZOEK,
+				icon: EDIT_ASSIGNMENT_BLOCK_ICONS()[AssignmentBlockType.ZOEK],
+				title: t('assignment/modals/add-block___zoeken-bouwen'),
+				description: t(
+					'assignment/modals/add-block___leer-leerlingen-zelf-bronnen-zoeken-of-laat-ze-een-collectie-samenstellen'
+				),
+				disabled:
+					assignment.blocks.findIndex(
+						(block) => block.type === AssignmentBlockType.ZOEK
+					) >= 0,
+			},
+			{
+				type: AssignmentBlockType.TEXT,
+				icon: EDIT_ASSIGNMENT_BLOCK_ICONS()[AssignmentBlockType.TEXT],
+				title: t('assignment/modals/add-block___instructies-tekst'),
+				description: t(
+					'assignment/modals/add-block___voeg-een-tekstblok-toe-met-instructies-of-wat-extra-informatie'
+				),
+			},
+		],
+		[assignment.blocks, t]
+	);
+
+	return (
+		<Modal
+			isOpen={isOpen}
+			title={t('assignment/modals/add-block___toevoegen')}
+			size="large"
+			onClose={onClose}
+			scrollable
+			className="c-content"
+		>
+			<ModalBody>
+				<Spacer margin={['top-small', 'bottom-small', 'left', 'right']}>
+					<ul className="c-add-block__list">
+						{items.map((item) => (
+							<li
+								className={[
+									'c-add-block__list-item',
+									...(item.disabled ? ['c-add-block__list-item--disabled'] : []),
+								].join(' ')}
+								key={item.type}
+							>
+								<div className="c-add-block__icon">
+									<Icon name={item.icon} />
+								</div>
+
+								<div className="c-add-block__content">
+									<BlockHeading type="h4">{item.title}</BlockHeading>
+
+									<p className="c-add-block__description">{item.description}</p>
+								</div>
+
+								<div className="c-add-block__actions">
+									{!item.disabled && (
+										<Button
+											icon="plus"
+											type="primary"
+											onClick={() => {
+												onConfirm && onConfirm(item.type);
+											}}
```

Comment:
QUESTION: Would it make sense to make the whole list item clickable, and not just the blue button on the right?
![image](https://user-images.githubusercontent.com/1710840/176147024-77dc2997-b666-443b-9361-dc0862af8909.png)

Reply:
Fixed in https://github.com/viaacode/avo2-client/pull/1367/commits/92d08f043dde87c80ab329c6edb2191311fa7205#diff-62beb8bca0acf85377a3c25e326fa73f10ae175231b9818b5e8102bb3c501b45